(PUP-3951) Default Windows API strings to UTF_8

Iristyle opened the pull request:
- Previously, only Windows API calls that were Registry related had
   the Encoding set to UTF_8 at their individual callsites.  Change the
   default from Encoding.default_external (which matched previous
   Puppet 3.7 and below behavior) to now favor UTF_8 instead.  Though
   this commit doesn't see those changes, the calls that will
   automatically pick up this new string encoding are:

   Puppet::Util::Windows::ADSI.computer_name
   Puppet::Util::Windows::ADSI::User.current_user_name
   Puppet::Util::Windows::Error.format_error_code
   Puppet::Util::Windows::File.resolve_symlink

puppetcla commented:
CLA signed by all contributors.

joshcooper commented:
@Iristyle I see this is targeted for 4.x, should we wait to merge? I assume the is super low risk and a house-cleaning thing to ensure we're consistent about strings we bring into ruby?

Iristyle commented:
@joshcooper I'm glad you didn't merge yet, as I forgot to `WIP` this one when I pushed up last night.

I have corrected the 1 failing test / repushed, and I retargeted the ticket at 4.0 b/c I think we should take the opportunity to ship these low-risk changes.  Really, the existing behavior doesn't make a lot of sense to try to convert `UTF-16LE` to `Default.external_encoding` given the potential for conversion related exceptions as we've seen with the Registry code.

I also added the ticket to the `Non US-Language / Unicode Support for Windows` epic.

Iristyle commented:
Feel free to merge this one - it's on the Windows board now and ready to go.

ferventcoder commented:
This is money.

ferventcoder commented:
I'm gonna give it a once over with specs tho

ferventcoder commented:
Passed specs locally, possibly still waiting on a good run from #3534 .
